### PR TITLE
Docker test execution support and logging 

### PR DESCRIPTION
**Pull request by agosh01**
This pull request focuses the end-to-end test execution process for streamer, zenoh, and vsomip. It introduces the capability for users to run test agents and streamer within Docker containers, and enables Test agents to send test results to the TM (Test Management) running outside docker.

Changes:
1. Run processes inside docket containers
2. Log subprocess output into log file under logs directory

### CODE REVIEW

File: test_manager/features/environment.py [hunk 1]
```diff
@@ -98,36 +48,21 @@ def before_all(context):
...
-    dispatcher = Dispatcher()
-    thread = Thread(target=dispatcher.listen_for_client_connections)
-    thread.start()
-    context.dispatcher = dispatcher
-
-    context.logger.info("Created Dispatcher...")
-    time.sleep(5)
-
-    test_manager = TestManager(context, "127.0.0.5", 12345)
+    test_manager = TestManager(context, "0.0.0.0", 12345)
```
Comment (matthewd0123) [inline test_manager/features/environment.py:59]: does it need to stay as 0.0.0.0?

Comment (agosh01) [reply test_manager/features/environment.py:59]: This is changing TM to global ip in case docker wants to connect. You don't have to change TA IPs

**Comment by neelam-kushwah** [comment]
@agosh01 , please resolve conflicts.

**Comment by agosh01** [comment]
> @agosh01 , please resolve conflicts.

@neelam-kushwah @matthewd0123 I am unable to see conflicting files. It says "Only those with [write access](https://docs.github.com/articles/what-are-the-different-access-permissions) to this repository can merge pull requests." 

Can you please check if this something which can granted to me ?

**Comment by agosh01** [comment]
> @agosh01 , please resolve conflicts.

@neelam-kushwah Done

File: test_manager/features/steps/tck_step_implementations.py [hunk 2]
```diff
@@ -50,6 +43,17 @@
 
 from dispatcher.dispatcher import Dispatcher
 
+def get_absolute_path(filepath_from_root_repo: str) -> str:
+    return os.path.abspath(os.path.dirname(os.getcwd()) + "/" + filepath_from_root_repo)
+
+def get_docker_host_ip():
+    cmd = "ip -4 addr show docker0 | grep -Po 'inet \\K[\\d.]+'"
```
Comment (princemac) [inline test_manager/features/steps/tck_step_implementations.py:50]: @agosh01 Isn't the command here, specific to linux environment?

Comment (agosh01) [reply test_manager/features/steps/tck_step_implementations.py:50]: yes, window will be likely virtual machine with docket which command will look like "docker-machine ip default". 

Can you provide more info about ask ? Do you have windows usecase for this ? 

I can add check for windows or linux and do appropriate execution

Comment (princemac) [reply test_manager/features/steps/tck_step_implementations.py:50]: Does current changes support running of docker container on linux and windows. Or only linux?

Comment (agosh01) [reply test_manager/features/steps/tck_step_implementations.py:50]: only linux but we do couple thing add check to restrict the user to run this only on linux or add windows compatibility if there is usecase for this

**Pull request closed without merging**